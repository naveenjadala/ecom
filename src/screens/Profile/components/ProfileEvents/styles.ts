import {StyleSheet} from 'react-native';
import {getFontSize} from '../../../../assets/fonts/FontSizes';
import {colors} from '../../../../assets/colors';

export const profileEventsStyles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    paddingHorizontal: '10%',
    paddingVertical: '5%',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  text: {
    color: colors.text,
    fontSize: getFontSize(16),
    letterSpacing: 0,
    textAlignVertical: 'center',
  },
  subText: {
    fontSize: getFontSize(13),
    color: colors.subtext,
  },
  icon: {
    alignSelf: 'center',
  },
});
