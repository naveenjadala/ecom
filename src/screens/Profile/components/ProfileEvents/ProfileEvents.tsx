import {View, Text, TouchableOpacity} from 'react-native';
import React, {FC} from 'react';
import {profileEventsStyles} from './styles';
import RightIcon from '../../../../assets/images/rightIcon.svg';

type profileEventsProps = {
  title: string;
  onClick: () => void;
  subTitle?: string;
};

const ProfileEvents: FC<profileEventsProps> = ({title, onClick, subTitle}) => {
  return (
    <TouchableOpacity
      accessibilityRole="button"
      onPress={() => console.log('click')}
      style={profileEventsStyles.container}>
      <View>
        <Text style={profileEventsStyles.text}>{title}</Text>
        <Text style={profileEventsStyles.subText}>{subTitle}</Text>
      </View>
      <RightIcon width={20} height={20} style={profileEventsStyles.icon} />
    </TouchableOpacity>
  );
};

export default ProfileEvents;
