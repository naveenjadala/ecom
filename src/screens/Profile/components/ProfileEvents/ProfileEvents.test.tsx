import {render, screen} from '../../../../test/test-utils';
import ProfileEvents from './ProfileEvents';

test('should render ProfileEvents snapShort', () => {
  render(
    <ProfileEvents
      title={'ProfileEvents'}
      onClick={function (): void {
        throw new Error('Function not implemented.');
      }}
    />,
  );
  expect(screen).toMatchSnapshot();
});

test('should render ProfileEvents element', () => {
  render(
    <ProfileEvents
      title={'ProfileEvents'}
      onClick={function (): void {
        throw new Error('Function not implemented.');
      }}
    />,
  );
  const getClickableElementView = screen.getByRole('button');
  expect(getClickableElementView).toBeTruthy();
});
