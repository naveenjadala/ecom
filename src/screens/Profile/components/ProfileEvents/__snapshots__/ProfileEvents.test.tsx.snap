// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProfileEvents snapShort 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "minHeight": "100%",
      "minWidth": "100%",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": "10%",
        "paddingVertical": "5%",
      }
    }
  >
    <View>
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 8,
            "letterSpacing": 0,
            "textAlignVertical": "center",
          }
        }
      >
        ProfileEvents
      </Text>
      <Text
        style={
          {
            "color": "#8C8C8C",
            "fontSize": 6.5,
          }
        }
      />
    </View>
    <test-file-stub
      height={20}
      style={
        {
          "alignSelf": "center",
        }
      }
      width={20}
    />
  </View>
</RCTSafeAreaView>
`;
